src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const newState = { ...state };
+  const stateResult = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        if (action.extraData && typeof action.extraData === 'object') {
+          Object.assign(newState, action.extraData);
+        }
+        break;
+      case 'removeProperties':
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        break;
+
+      case 'clear':
+        for (const object in newState) {
+          delete newState[object];
+        }
+        break;
+    }
+    stateResult.push({ ...newState });
```

Comment:
The operation of pushing a copy of `newState` should be moved outside the `switch` block. This will ensure that the state is consistently updated after each action.